src/cascadia/TerminalApp/HighlightedTextControl.h

```diff
@@ -0,0 +1,56 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+#pragma once
+
+#include "winrt/Microsoft.UI.Xaml.Controls.h"
+
+#include "HighlightedTextSegment.g.h"
+#include "HighlightedText.g.h"
+#include "HighlightedTextControl.g.h"
+#include "../../cascadia/inc/cppwinrt_utils.h"
+
+namespace winrt::TerminalApp::implementation
+{
+    struct HighlightedTextSegment : HighlightedTextSegmentT<HighlightedTextSegment>
```

Comment:
I'm 50/50 split on whether I think these three should be in their own files or not. They're tightly coupled enough that I guess it's sensible for them all to be in the same file...